### ISSUE TITLE
maintainBrowserTab configuration option

### ISSUE THREAD

**Issue by kasp1** [apiko-rest-api/apiko#56]
Originally I was thinking that the server side of Apiko could open a browser tab on startup with the dev UI in development mode, which would make it easier to get started with Apiko in general.

Then someone mentioned that Apiko could have a CLI tool to make things easier, so I've made it too see how will it turn out.

The problem with Apiko CLI is that it doesn't know when the server runs in "development" environment and when does it run in "production". Because environments are not strictly defined in Apiko, anyone can choose their way to define environments, here's what it looks like in the starter template for example:

```
if (process.argv[2] === 'prod') {
  // protects the server with the secret generated on the first run (see apiko.json)
}

if (process.argv[2] === 'dev') {
  config.protect = false, // if true, protects /dev/ UI with the secret generated on the first run (see apiko.json), use for production
  config.verbosity = 2 // 0 - 3
}
```

As you can see, a configuration set for an environment may have any name the user may choose. So the Apiko CLI tool can't predict which one is "development". Given the main file included in package.json, the most simplicity Apiko CLI can provide as of now is: `apiko run <env_name>`.

So perhaps maintaining the browser tab should be really up to the Apiko server application. We could possibly add a configuration option like `maintainBrowserTab`, which the user could use in their definition of "development" environment.

Finally, the `apiko setup <directory>` command downloads the starter repo, where the development environment is initially named "dev". So this "setup" command can be sure to use it. With the new installation instructions, a newcomer will have to only run this "setup" command and wait a while, then the dev UI will pop up to their face with no additional hassle.

**Comment by ilabacheuski** [comment]
Since apiko is node application why not to use NODE_ENV?
And it can be accessed by [process.env](https://nodejs.org/api/process.html#process_process_env) 
Simple and common practic: [Express](https://github.com/expressjs/express/search?utf8=✓&q=process.env&type=), [Sequelize-CLI](https://github.com/sequelize/cli/search?utf8=✓&q=process.env&type=)

**Comment by kasp1** [comment]
What could the startup command (`apiko run <environment>`) look like if we used process.env? How would it work?

**Comment by ilabacheuski** [comment]
Setting NODE_ENV:
```bash
linux & mac: export NODE_ENV=production
windows CMD (not PS!): set NODE_ENV=production
```
Production: `apiko run`
Development `set NODE=development && apiko run`
Testing`set NODE=test && apiko run` something like it. And for cross-platform setting there is plugin [cross-env](https://github.com/kentcdodds/cross-env)
So the deployment option is stored in user environment variable NODE_ENV(we can name it differently of course). And in CLI you just access it by `process.env.NAME_OF_NEEDED_VARIABLE`

**Comment by kasp1** [comment]
Okay, let's keep it in mind. We'll still have this option in the config though, right?

**Comment by ilabacheuski** [comment]
Yeah. If we need it. In Production you don't need this option at all. That's why NODE_ENV is usually used.
In package.json is possible to have scripts `run, dev, test` and so on.

**Comment by kasp1** [comment]
I realized that if someone runs `apiko setup <project_directory>` then the starter template is downloaded and it always contains the starter configuration where `dev` is always the development environment and that gets start up.

Theoretically we could open the browser tab just for the `apiko setup <...>` command. Just for the newbies so they don't get confused about no UI appearing. This way we could implement it in the CLI tool.

Thinking...
